lib/p5.fab.js

```diff
@@ -258,376 +269,463 @@ class Fab {
-        this.add(cmd);
-        this.print();
-    }
-
-    introLine(z = 0.3) {
-        this.move(0.1, 20, z, 85);
-        this.moveExtrude(0.1, 200, z, 25);
-        this.addComment('intro');
-        this.move(0.4, 200.0, z, 85);
-        this.moveExtrude(0.4, 20.0, z, 25);
-        this.addComment('intro');
-        // adding header from cura intro
-        this.add("G92 E0");
-        this.add("G0 Z2.0 F3000");
-        this.add("G0 X5 Y20 Z0.3 F5000.0");
-        this.add("G92 E0");
-    }
-
-    presentPart() {
-        var retractCmd = "G1 E-10.0 F6000";
-        this.add(retractCmd);
-        var cmd = "G0 X0 Y180 F9000";
-        this.add(cmd);
-    }
-
-    waitCommand() {
-        var cmd = "M400";
-        this.add(cmd);
-    }
-
-    getPos() {
-        var cmd = "M114_DETAIL";
-        var cmd = "M114 D";
-        this.add(cmd);
-    }
-
-    setPos() {
-        var cmd = `G92 X${this.x} Y${this.y} Z${this.z} E${this.e}`
-    }
-
-    autoReportPos(t = 10) {
-        // currently not working
-        this.add("AUTO_REPORT_POSITION");
-        t = parseInt(t);
-        var cmd = `M154 S${t}`;
-        this.add(cmd);
-    }
-
-    /*****
-    * G-Code Path Commands
-    */
-    move(x, y, z, v) {
-        this.x = parseFloat(x).toFixed(2);
-        this.y = parseFloat(y).toFixed(2);
-        this.z = parseFloat(z).toFixed(2);
-        v = this.mm_sec_to_mm_min(v);
-        this.v = parseFloat(v).toFixed(2);
-
-        var cmd = `G0 X${this.x} Y${this.y} Z${this.z} F${this.v}`;
-        this.add(cmd);
-
-        return cmd;
-    }
-
-    moveX(x, v = 25) {
-        this.x = parseFloat(x).toFixed(2);
-        v = this.mm_sec_to_mm_min(v);
-        this.v = parseFloat(v).toFixed(2);
-
-        var cmd = `G0 X${this.x} F${this.v}`;
-        this.add(cmd);
-
-        return cmd;
-    }
-
-    moveY(y, v = 25) {
-        this.y = parseFloat(y).toFixed(2);
-        v = this.mm_sec_to_mm_min(v);
-        this.v = parseFloat(v).toFixed(2);
-
-        var cmd = `G0 Y${this.y} F${this.v}`;
-        this.add(cmd);
-
-        return cmd;
-    }
-
-    moveZ(z, v = 25) {
-        this.z = parseFloat(z).toFixed(2);
-        v = this.mm_sec_to_mm_min(v);
-        this.v = parseFloat(v).toFixed(2);
-
-        var cmd = `G0 Z${this.z} F${this.v}`;
-        this.add(cmd);
-
-        return cmd;
-    }
-
-    up(z, v = 50) {
-        this.z = parseFloat(z).toFixed(2);
-        v = this.mm_sec_to_mm_min(v);
-        this.v = parseFloat(v).toFixed(2);
-
-        var cmd = `G0 Z${this.z} F${this.v}`;
-        this.add(cmd);
-
-        return cmd;
-    }
-
-    // moveExtrude(x, y, z, v = 25, e = this.makeE(x, y, z)) {
-    moveExtrude(x, y, z, v = 25, e = null, multiplier = false) {
-        // this if statement was added
-        if (e == null) {
-            this.e = this.makeE(x, y, z);
-        }
-        else if (multiplier) {
-            this.e = e * this.makeE(x, y, z);
+  render() {
+    if (this.coordinateSystem == "delta") {
```

Comment:
the way the model is placed on the bed does not differ between the two printer types. The only difference is how the printer is drawn